newlib/libc/time/mktime.c

```diff
@@ -77,6 +78,13 @@ static const int _DAYS_BEFORE_MONTH[12] =
 #define _ISLEAP(y) (((y) % 4) == 0 && (((y) % 100) != 0 || (((y)+1900) % 400) == 0))
 #define _DAYS_IN_YEAR(year) (_ISLEAP(year) ? 366 : 365)
 
+static void
+set_tm_wday (long days, struct tm *tim_p)
+{
+  if ((tim_p->tm_wday = (days + 4) % 7) < 0)
```

Comment:
Please move the assignment to a separate statement instead of placing it inside the 'if' conditional.

Reply:
I actually agree with you here. But that piece of code is exactly like the original, I only moved it to a new routine. The strategy I followed is to change the original code as little as possible (one of the options you already mentioned in an e-mail), in order to make the review easier. This strategy could also increase the chances of getting this patch into the main Newlib. So I would rather not change it now. I am planning more changes to this file, so I'll add that to the list of candidates.